flask/flask-common/src/main/java/net/corda/flask/common/LockFile.java

```diff
@@ -17,7 +17,6 @@
 
     private static RandomAccessFile createLockFile(Path path) throws IOException {
         Files.createDirectories(path.getParent());
```

Comment:
I think [`Files.createDirectories`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createDirectories-java.nio.file.Path-java.nio.file.attribute.FileAttribute...-) will also throw `FileAlreadyExistsException` if two threads race to create the same directories.

Reply:
in theory, [it shouldn't](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createDirectories-java.nio.file.Path-java.nio.file.attribute.FileAttribute...-)
```
Unlike the createDirectory method, an exception is not thrown if the directory could not be created because it already exists
```

Reply:
Yes, I saw that. Hence why I updated my comment.